BH/Modules/Item/ItemDisplay.cpp

```diff
@@ -2341,7 +2341,7 @@ bool MapIdCondition::EvaluateInternal(UnitItemInfo* uInfo,
 	Condition* arg1,
 	Condition* arg2)
 {
-	auto map_id = **Var_D2CLIENT_MapId();
+	int map_id = (int)D2CLIENT_GetPlayerUnit()->pAct->pRoom1->pRoom2->pLevel->dwLevelNo;

```

Comment:
This will need some safety NULL checks, not all of these are guaranteed to be defined at all times and could cause client crashes from null references